pyteal/stack_frame.py

```diff
@@ -318,46 +331,96 @@ def _best_frame_as_pyteal_frame(self) -> "PyTealFrame | None":
         return None
 
     def __repr__(self) -> str:
-        return f"{'C' if self._compiler_gen else 'U'}{self._frames}"
+        return f"NatalStackFrame({self._frames=})"
 
     def nodes(self) -> list[AST | None]:
         return [f.node for f in self._frames]
 
     @classmethod
-    def _walk_asts(cls, func: Callable[["Expr"], None], *exprs: "Expr") -> None:  # type: ignore
+    def _walk_asts(
+        cls,
+        func: Callable[["Expr"], bool],  # type: ignore[name-defined]
+        *exprs: "Expr",  # type: ignore[name-defined]
+        force_root_apply: bool = True,
+        exit_on_user_defined: bool = False,
+        exit_when_stop_signal: bool = False,
+    ) -> None:
```

Comment:
nit: I guess I understood the `func` is a method that traverse AST while giving out some auxiliary information on if keep traversing.

Maybe a sentence or two on commenting the purpose of the following configs would be helpful, in case my memory on the code fades away:
- `func`
- `force_root_apply`
- `exit_on_user_defined`
- `exit_when_stop_signal`

Reply:
622bb848ebffa6b8e81dbecde58338e90ff29793